src/master.py

```diff
@@ -116,13 +117,13 @@ def remaster(stackname, new_master_stackname):
         except KeyboardInterrupt:
             LOG.info("not skipping")
 
-    LOG.info("setting new master address")
-    context = cfngen.set_master_address(context, master_ip) # mutator
-
     LOG.info("upgrading salt client")
     pdata = core.project_data_for_stackname(stackname)
     context['project']['salt'] = pdata['salt']
 
+    LOG.info("setting new master address")
+    cfngen.set_master_address(pdata, context, master_ip) # mutator
```

Comment:
not clear in general what it mutates (`context` as opposed to `pdata`), could be the first argument

Reply:
I'll elaborate